src/sentry/tasks/integrations/github/pr_comment.py

```diff
@@ -318,3 +325,43 @@ def github_comment_reactions():
             continue
 
         metrics.incr("github_pr_comment.comment_reactions.success")
+
+
+# TODO(adas): Change the client typing to allow for multiple SCM Integrations
+def safe_for_comment(
+    gh_client: GitHubAppsClient, repository: Repository, pull_request: PullRequest
+) -> bool:
+    try:
+        pullrequest_resp = gh_client.get_pullrequest(
+            repo=repository.name, pull_number=pull_request.key
+        )
+    except ApiError as e:
+        if e.json and RATE_LIMITED_MESSAGE in e.json.get("message", ""):
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "gh_rate_limited", "code": e.code},
+            )
+        elif e.code == 404:
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "missing_gh_pull_request", "code": e.code},
+            )
+        else:
+            metrics.incr(
+                OPEN_PR_METRIC_BASE.format(key="api_error"),
+                tags={"type": "unknown_api_error", "code": e.code},
+            )
+        raise e
+
+    safe_to_comment = True
+    if pullrequest_resp["changed_files"] > OPEN_PR_MAX_FILES_CHANGED:
+        metrics.incr(
+            OPEN_PR_METRIC_BASE.format(key="rejected_comment"), tags={"reason": "too_many_files"}
+        )
+        safe_to_comment = False
```

Comment:
Why not just `return False` here and in the case below and `return True` at the end if it hasn't exited?

Reply:
The check could fail for two reasons and I want to increment both metrics if both checks fail before exiting.

Reply:
makes sense!